From 69f289964dd9457188ffa13cfb592ecde8b42375 Mon Sep 17 00:00:00 2001
From: Basanagouda Koppad <Basanagoudax.n.Koppad@intel.com>
Date: Mon, 8 Mar 2021 09:25:31 +0530
Subject: [PATCH] allow appdomain to access priv_app_tmpfs

Some CTS testcases need to access contact app/service,
which is a priveleged module. These cases need these rules
to complete their work.

Tracked-On: OAM-96364
Signed-off-by: ji, zhenlong z <zhenlong.z.ji@intel.com>
Signed-off-by: Basanagouda Koppad <Basanagoudax.n.Koppad@intel.com>

diff --git a/prebuilts/api/30.0/private/app.te b/prebuilts/api/30.0/private/app.te
index c8392eedf..0a7c78b8b 100644
--- a/prebuilts/api/30.0/private/app.te
+++ b/prebuilts/api/30.0/private/app.te
@@ -22,6 +22,8 @@ dontaudit appdomain system_data_file:dir write;
 # Apps should not be reading vendor-defined properties.
 dontaudit appdomain vendor_default_prop:file read;

+allow appdomain priv_app_tmpfs:file read;
+
 neverallow appdomain system_server:udp_socket {
         accept append bind create ioctl listen lock name_bind
         relabelfrom relabelto setattr shutdown };
diff --git a/prebuilts/api/30.0/private/priv_app.te b/prebuilts/api/30.0/private/priv_app.te
index 44c81ee80..0f319e102 100644
--- a/prebuilts/api/30.0/private/priv_app.te
+++ b/prebuilts/api/30.0/private/priv_app.te
@@ -2,9 +2,13 @@
 ### A domain for further sandboxing privileged apps.
 ###

+type priv_app_tmpfs, file_type;
+
 typeattribute priv_app coredomain;
 app_domain(priv_app)

+typeattribute priv_app_tmpfs mlstrustedobject;
+
 # Access the network.
 net_domain(priv_app)
 # Access bluetooth.
diff --git a/private/app.te b/private/app.te
index c8392eedf..0a7c78b8b 100644
--- a/private/app.te
+++ b/private/app.te
@@ -22,6 +22,8 @@ dontaudit appdomain system_data_file:dir write;
 # Apps should not be reading vendor-defined properties.
 dontaudit appdomain vendor_default_prop:file read;

+allow appdomain priv_app_tmpfs:file read;
+
 neverallow appdomain system_server:udp_socket {
         accept append bind create ioctl listen lock name_bind
         relabelfrom relabelto setattr shutdown };
diff --git a/private/priv_app.te b/private/priv_app.te
index 44c81ee80..0f319e102 100644
--- a/private/priv_app.te
+++ b/private/priv_app.te
@@ -2,9 +2,13 @@
 ### A domain for further sandboxing privileged apps.
 ###

+type priv_app_tmpfs, file_type;
+
 typeattribute priv_app coredomain;
 app_domain(priv_app)

+typeattribute priv_app_tmpfs mlstrustedobject;
+
 # Access the network.
 net_domain(priv_app)
 # Access bluetooth.
--
2.30.1

